### PR TITLE
Cartoframes viz show

### PR DESCRIPTION
**Pull request by ethervoid**
Related https://github.com/CartoDB/cartodb/issues/14899

[Cache platform PR](https://github.com/CartoDB/cartodb-platform/pull/5671)

### CODE REVIEW

File: app/controllers/carto/kuviz/visualizations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        render 'show', layout: false
+      rescue => e
+        CartoDB::Logger.error(exception: e)
+        kuviz_password_protected
+      end
+
+      private
+
+      def get_kuviz
+        @kuviz = Carto::Asset.find_by_visualization_id(params[:id])
```
[the remark this answers is not in the document]
Comment (ethervoid) [reply app/controllers/carto/kuviz/visualizations_controller.rb]: I've change the model relation [here](https://github.com/CartoDB/cartodb/pull/14918/files#diff-4cab6f44c5314b6952a26fe07bdd1d73R81) so I'm going to change it in that branch.

File: app/controllers/carto/kuviz/visualizations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+      before_action :get_kuviz
+      skip_before_filter :verify_authenticity_token, only: [:show_protected]
+
+      def show
+        return kuviz_password_protected if @kuviz.visualization.password_protected?
+        @source = read_source_data
+        add_cache_headers
+        render layout: false
+      rescue => e
```
[the remark this answers is not in the document]
Comment (ethervoid) [reply app/controllers/carto/kuviz/visualizations_controller.rb]: Yes I missed that one

File: app/controllers/carto/kuviz/visualizations_controller.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        render 'kuviz_password', layout: 'application_password_layout'
+      end
+
+      def add_cache_headers
+        response.headers['X-Cache-Channel'] = "#{@kuviz.visualization.varnish_key}:vizjson"
+        response.headers['Surrogate-Key'] = "#{CartoDB::SURROGATE_NAMESPACE_PUBLIC_PAGES} #{@kuviz.visualization.surrogate_key}"
+        response.headers['Cache-Control'] = "no-cache,max-age=86400,must-revalidate,public"
+      end
+
+      def render_pretty_404
```
[the remark this answers is not in the document]
Comment (ethervoid) [reply app/controllers/carto/kuviz/visualizations_controller.rb]: Removed

File: app/controllers/carto/kuviz/visualizations_controller.rb [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        kuviz_password_protected
+      end
+
+      private
+
+      def get_kuviz
+        @kuviz = Carto::Asset.find_by_visualization_id(params[:id])
+      end
+
+      def read_source_data
```
[the remark this answers is not in the document]
Comment (ethervoid) [reply app/controllers/carto/kuviz/visualizations_controller.rb]: Yes it makes sense, I thought about it in the implementation but I didn't see the place to put it. Asset doesn't work that way, all the low level methods are private, but could make sense to move it to KuvizAssetsService.

**Comment by ethervoid** [comment]
@gonzaloriestra changes done

**Comment by gonzaloriestra** [comment]
retest this please

**Pull request merged**